src/sqltest/services/connectionManagement/connectionManagementService.test.ts

```diff
@@ -0,0 +1,170 @@
+import { ConnectionProfile } from 'sql/platform/connection/common/connectionProfile';
+import { CapabilitiesTestService } from 'sqltest/stubs/capabilitiesTestService';
+import { ConnectionOptionSpecialType, ServiceOptionType } from 'sql/workbench/api/common/sqlExtHostTypes';
+import { ConnectionProfileGroup } from 'sql/platform/connection/common/connectionProfileGroup';
+
+const sqlProvider = {
+	providerId: 'MSSQL',
+	displayName: 'MSSQL',
+	connectionOptions: [
+		{
+			name: 'connectionName',
+			displayName: undefined,
+			description: undefined,
+			groupName: undefined,
+			categoryValues: undefined,
+			defaultValue: undefined,
+			isIdentity: true,
+			isRequired: true,
+			specialValueType: ConnectionOptionSpecialType.connectionName,
+			valueType: ServiceOptionType.string
+		},
+		{
+			name: 'serverName',
+			displayName: undefined,
+			description: undefined,
+			groupName: undefined,
+			categoryValues: undefined,
+			defaultValue: undefined,
+			isIdentity: true,
+			isRequired: true,
+			specialValueType: ConnectionOptionSpecialType.serverName,
+			valueType: ServiceOptionType.string
+		},
+		{
+			name: 'databaseName',
+			displayName: undefined,
+			description: undefined,
+			groupName: undefined,
+			categoryValues: undefined,
+			defaultValue: undefined,
+			isIdentity: true,
+			isRequired: true,
+			specialValueType: ConnectionOptionSpecialType.databaseName,
+			valueType: ServiceOptionType.string
+		},
+		{
+			name: 'userName',
+			displayName: undefined,
+			description: undefined,
+			groupName: undefined,
+			categoryValues: undefined,
+			defaultValue: undefined,
+			isIdentity: true,
+			isRequired: true,
+			specialValueType: ConnectionOptionSpecialType.userName,
+			valueType: ServiceOptionType.string
+		},
+		{
+			name: 'authenticationType',
+			displayName: undefined,
+			description: undefined,
+			groupName: undefined,
+			categoryValues: undefined,
+			defaultValue: undefined,
+			isIdentity: true,
+			isRequired: true,
+			specialValueType: ConnectionOptionSpecialType.authType,
+			valueType: ServiceOptionType.string
+		},
+		{
+			name: 'password',
+			displayName: undefined,
+			description: undefined,
+			groupName: undefined,
+			categoryValues: undefined,
+			defaultValue: undefined,
+			isIdentity: true,
+			isRequired: true,
+			specialValueType: ConnectionOptionSpecialType.password,
+			valueType: ServiceOptionType.string
+		},
+		{
+			name: 'encrypt',
+			displayName: undefined,
+			description: undefined,
+			groupName: undefined,
+			categoryValues: undefined,
+			defaultValue: undefined,
+			isIdentity: false,
+			isRequired: false,
+			specialValueType: undefined,
+			valueType: ServiceOptionType.string
+		}
+	]
+};
+
+const capabilitiesService = new CapabilitiesTestService();
+capabilitiesService.capabilities['MSSQL'] = { connection: sqlProvider };
+
+suite('ConnectionManagementService Tests:', () => {
+	test('getConnections test', () => {
```

Comment:
>getConnections test [](start = 7, length = 19)

It's not a huge deal - but I really prefer to split tests like this into separate tests for each bit of functionality. That was you can easily see from running the tests what is being tested and it's much easier to debug issues since you don't have to wade through a ton of other code